go-controller/pkg/node/secondary_node_network_controller.go

```diff
@@ -28,15 +28,30 @@ type SecondaryNodeNetworkController struct {
 
 // NewSecondaryNodeNetworkController creates a new OVN controller for creating logical network
 // infrastructure and policy for default l3 network
-func NewSecondaryNodeNetworkController(cnnci *CommonNodeNetworkControllerInfo, netInfo util.NetInfo) *SecondaryNodeNetworkController {
-	return &SecondaryNodeNetworkController{
+func NewSecondaryNodeNetworkController(cnnci *CommonNodeNetworkControllerInfo, netInfo util.NetInfo, defaultNetworkGateway Gateway) (*SecondaryNodeNetworkController, error) {
+	snnc := &SecondaryNodeNetworkController{
 		BaseNodeNetworkController: BaseNodeNetworkController{
 			CommonNodeNetworkControllerInfo: *cnnci,
 			NetInfo:                         netInfo,
 			stopChan:                        make(chan struct{}),
 			wg:                              &sync.WaitGroup{},
 		},
 	}
+	if util.IsNetworkSegmentationSupportEnabled() && snnc.IsPrimaryNetwork() {
+		node, err := snnc.watchFactory.GetNode(snnc.name)
+		if err != nil {
+			return nil, err
+		}
+		networkID, err := snnc.getNetworkID()
```

Comment:
This net ID is set async right - in-order to try to reduce log spam, do you see this being a problem? Should we exit with no error when net ID isnt set yet?

Reply:
also enrich the errors in this func. Dont want to see `Failed to create network xyz: object not found`

Reply:
done for enriching error msg